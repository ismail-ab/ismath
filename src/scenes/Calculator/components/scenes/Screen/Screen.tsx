import * as React from 'react';
import { Typography } from '@material-ui/core';
import { CalculatorContext } from '../../../contexts';

const Screen: React.SFC = () => (
  <CalculatorContext.Consumer>
    {({ concatenatedCharacters, result }) => (
      <>
        <Typography
          color="primary"
          variant="headline"
          component="h2"
          align="right"
          style={{ height: '40px' }}
        >
          {result}
        </Typography>
        <Typography
          color="secondary"
          component="h2"
          align="right"
          noWrap={true}
          style={{
            height: '25px',
            fontSize: '1.2em'
          }}
        >
          {concatenatedCharacters}
        </Typography>
      </>
    )}
  </CalculatorContext.Consumer>
);

export default Screen;