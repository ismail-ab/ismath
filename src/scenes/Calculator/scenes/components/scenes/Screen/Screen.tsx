import * as React from 'react';
import { Typography } from '@material-ui/core';

interface IScreenOwnProps {
  concatenatedCharacters: string;
  result: number | string;
}

const Screen: React.SFC<IScreenOwnProps> = ({ concatenatedCharacters, result }) => (
  <>
    <Typography
      color="primary"
      variant="headline"
      component="h2"
      align="right"
      style={{ height: '40px' }}
    >
      {result}
    </Typography>
    <Typography
      color="secondary"
      component="h2"
      align="right"
      noWrap={true}
      style={{
        height: '25px',
        fontSize: '1.2em'
      }}
    >
      {concatenatedCharacters}
    </Typography>
  </>
);

export default Screen;