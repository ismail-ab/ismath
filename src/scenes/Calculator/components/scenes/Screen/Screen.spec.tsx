import { } from 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';
import Screen from './Screen';

describe('Screen', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Screen />);

    expect(wrapper).toMatchSnapshot();
  });
});
