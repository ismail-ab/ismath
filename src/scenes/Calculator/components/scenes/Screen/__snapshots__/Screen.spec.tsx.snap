// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen renders correctly 1`] = `<[object Object] />`;
